app/main.py

```diff
@@ -1,34 +1,79 @@
 class Deck:
     def __init__(self, row, column, is_alive=True):
-        pass
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
     def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = []
+
+    def decks_create(self):
+        list_start = list(self.start)
+        list_end = list(self.end)
+
+        if list_start == list_end:
+            self.decks.append(Deck(list_start[0], list_start[1]))
+
+        if list_start[0] != list_end[0]:
+            for points in range(list_start[0], list_end[0] + 1):
+                self.decks.append(Deck(points, list_start[1]))
+
+        if list_start[1] != list_end[1]:
+            for points in range(list_start[1], list_end[1] + 1):
+                self.decks.append(Deck(list_start[0], points))
+
+        self.decks.append(self.is_drowned)
+        return self.decks
 
     def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        for deck in range(0, len(self.decks) - 1):
+            if self.decks[deck].row == row and \
+                    self.decks[deck].column == column:
+                return Deck(row, column)
 
     def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+        for deck in range(0, len(self.decks) - 1):
+            if self.decks[deck].row == row and \
+                    self.decks[deck].column == column:
+                self.decks[deck].is_alive = False
+        list = []
+        for decks in range(0, len(self.decks) - 1):
+            list.append(self.decks[decks].is_alive)
+        if sum(list) == 0:
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
+    def __init__(self, ships, field={}):
+        self.ships = ships
+        self.field = field
+        self.field = Battleship.create(self)
+
+    def create(self):
+        list_ships = []
+        for ships in self.ships:
+            battle_ships = list(ships)
+            list_ships.append(Ship(battle_ships[0], battle_ships[1]))
+        for create_decks in list_ships:
+            create_decks.decks_create()
+        for crate_field in list_ships:
+            for j in range(len(crate_field.decks) - 1):
+                self.field.update({(crate_field.decks[j].row,
```

Comment:
```suggestion
                self.field.update[(crate_field.decks[j].row, crate_field.decks[j].column] = crate_field
```
This way of creating a pair in dict is more compact

Reply:
TypeError: 'builtin_function_or_method' object does not support item assignment

Reply:
Sorry, forgot to delete `update` word:
```
 self.field[(crate_field.decks[j].row, crate_field.decks[j].column] = crate_field
```